File: implementations/react/src/patterns/behavioral/command/add-item-command.ts
import type { Command } from "./command-interface";
import type { Receiver } from "./receiver";

/**
 * Comando concreto para agregar un ítem en el patrón Command.
 */

export class AddItemCommand implements Command {
    private device: Receiver;

    constructor(device: Receiver){
        this.device = device;
    }

    /** Ejecuta el comando para agregar un ítem */
    execute(item: string) {
        this.device.addItem(item);
    }
}

File: implementations/flutter/design_patterns_flutter/lib/patterns/behavioral/command/add_item_command.dart
import 'package:design_patterns_flutter/patterns/behavioral/command/command.dart';
import 'package:design_patterns_flutter/patterns/behavioral/command/receiver.dart';

class AddItemCommand implements Command {
  final Receiver device;
  AddItemCommand(this.device);

  @override
  void execute([String? item]) {
    if (item != null && item.isNotEmpty) {
      device.addItem(item);
    }
  }
}
